digitalidentity/service.go

```diff
@@ -145,3 +152,120 @@ func GetShareQr(httpClient requests.HttpClient, sessionID string, clientSdkId, a
 
 	return share, err
 }
+
+// Get Receipt info using the supplied receiptId
+func GetReceipt(httpClient requests.HttpClient, receiptId string, clientSdkId, apiUrl string, key *rsa.PrivateKey) (share ReceiptResponse, err error) {
+	receiptUrl := requests.Base64ToBase64URL(receiptId)
+	endpoint := fmt.Sprintf(identitySessionReceiptRetrieval, receiptUrl)
+
+	headers := requests.AuthHeader(clientSdkId)
+	request, err := requests.SignedRequest{
+		Key:        key,
+		HTTPMethod: http.MethodGet,
+		BaseURL:    apiUrl,
+		Endpoint:   endpoint,
+		Headers:    headers,
+	}.Request()
+	if err != nil {
+		return share, err
+	}
+
+	response, err := requests.Execute(httpClient, request)
+	if err != nil {
+		return share, err
+	}
+	defer response.Body.Close()
+
+	responseBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return share, err
+	}
+
+	err = json.Unmarshal(responseBytes, &share)
+
+	return share, err
+}
+
+// Get Receipt info using the supplied receiptId
+func GetFetchReceiptItemKey(httpClient requests.HttpClient, fetchReceiptItemKeyId string, clientSdkId, apiUrl string, key *rsa.PrivateKey) (share ReceiptItemKeyResponse, err error) {
+
+	endpoint := fmt.Sprintf(identitySessionReceiptKeyRetrieval, fetchReceiptItemKeyId)
+	headers := requests.AuthHeader(clientSdkId)
+	request, err := requests.SignedRequest{
+		Key:        key,
+		HTTPMethod: http.MethodGet,
+		BaseURL:    apiUrl,
+		Endpoint:   endpoint,
+		Headers:    headers,
+	}.Request()
+	if err != nil {
+		return share, err
+	}
+
+	response, err := requests.Execute(httpClient, request)
+	if err != nil {
+		return share, err
+	}
+	defer response.Body.Close()
+
+	responseBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return share, err
+	}
+
+	err = json.Unmarshal(responseBytes, &share)
+
+	return share, err
+}
+
+func GetShareReceipt(httpClient requests.HttpClient, receiptId string, clientSdkId, apiUrl string, key *rsa.PrivateKey) (share SharedReceiptResponse, err error) {
+	receiptResponse, err := GetReceipt(httpClient, receiptId, clientSdkId, apiUrl, key)
+
+	itemKeyId := receiptResponse.WrappedItemKeyId
+
+	encryptedItemKeyResponse, err := GetFetchReceiptItemKey(httpClient, itemKeyId, clientSdkId, apiUrl, key)
+
+	receiptContentKey, err := cryptoutil.UnwrapReceiptKey(receiptResponse.WrappedKey, encryptedItemKeyResponse.Value, encryptedItemKeyResponse.Iv, key)
+
+	//App Content
+	aattr, err := cryptoutil.DecryptReceiptContent([]byte(receiptResponse.Content.Profile), receiptContentKey)
+	aextra, err := cryptoutil.DecryptReceiptContent([]byte(receiptResponse.Content.ExtraData), receiptContentKey)
+
+	attrData := &yotiprotoattr.AttributeList{}
+	if err := proto.Unmarshal(aattr, attrData); err != nil {
+		return share, err
+	}
+
+	applicationProfile := newApplicationProfile(attrData)
+	extraDataValue, err := extra.NewExtraData(aextra)
+	if err != nil {
+		return share, err
+	}
+
+	applicationContent := ApplicationContent{applicationProfile, extraDataValue}
```

Comment:
@mehmet-yoti we still need to do this, the problem is i can't suggest which one because i'm not actually sure what's expected to be required or optional so i'd suggest having a chat with connect guys to see if `content` or `otherPartyContent` can ever be `nil` and if `profile` or `extraData` can ever be empty in `content` or `otherPartyContent` (the `omitempty` in our response structs suggest it can be).

IF they don't know either and we can't get an answer, the safe option is just the second suggestion of only doing things if it's `not nil` or `not empty`. This is a principle we need to always keep in mind in go code to just protect ourselves (the RPs in this case) from panics in the server and unexpected errors